### PR TITLE
(814b) Switch endpoint for fetching user's caseloads

### PR DESCRIPTION
**Pull request by Gweaton**
## Context

The previous endpoint (`/api/staff/:staffId/caseloads`) wouldn't return any caseloads for admin users (e.g. my own). It also meant we had to do a slightly roundabout journey reading the logged-in-user's staffID.

## Changes in this PR

Switches to using the (apparently case-sensitive) `/api/users/me/caseLoads` endpoint and passing the logged-in user's token. This seems to allow us to fetch caseloads even if the user is an admin.

We'll be passing these caseloads to the request to the Prisoner Offender Search for a prisoner to ensure we're only searching by the logged in user's caseloads.

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/paths/prisonApi.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { path } from 'static-path'
 
-const caseloadsPath = path('/api/staff/:staffId/caseloads')
+const caseloadsPath = path('/api/users/me/caseLoads')
```
Comment (Gweaton) [inline server/paths/prisonApi.ts:3]: That's not typo in `caseLoads` 😬

Comment (yndajas) [reply server/paths/prisonApi.ts:3]: What an L

Comment (yndajas) [reply server/paths/prisonApi.ts:3]: So the endpoint uses `me`? I think it might still be nicer to use `current`/`currentUser` internally

**Review by yndajas — APPROVED** [review]

File: server/paths/prisonApi.ts (hunk 1 above)
Comment (Gweaton) [reply server/paths/prisonApi.ts:3]: how do you mean? The Prison API endpoint uses "me" (we have no control over this), but internally we're referring to it as `current`

**Pull request merged**